Userland/Applications/Piano/AudioPlayerLoop.cpp

```diff
@@ -139,10 +140,12 @@ void AudioPlayerLoop::write_wav_if_needed()
             m_track_manager.reset();
             m_track_manager.set_should_loop(false);
             do {
+                // FIXME: This progress detection is crude, but it works for now.
+                m_wav_percent_written.store(static_cast<int>(static_cast<float>(m_track_manager.transport()->time()) / roll_length * 100.0f));
```

Comment:
Atomics are surprisingly slow, and this looks like a hot loop. Have you tried storing the percent in a local, and only writing it to the atomic when it actually changes?

Reply:
The amount of time spent on actually writing data to disk is significantly higher. Also, we write in ~page sized blocks, so we only write the atomic about ten times per second.